rosplan_planning_system/src/PlanDispatch/EsterelPlanDispatcher.cpp

```diff
@@ -3,396 +3,510 @@
-					dest << "\",style=filled,fillcolor=darkgoldenrod2];" << std::endl;
-				} else {
-					dest << "\"];" << std::endl;
-				}
-				break;
-			case rosplan_dispatch_msgs::EsterelPlanNode::PLAN_START:
-				dest << "\",style=filled,fillcolor=black,fontcolor=white];" << std::endl;
-				break;
-			default:
-				dest << "\"];" << std::endl;
-				break;
-			}
-		}
-
-		// edges
-		for(std::vector<rosplan_dispatch_msgs::EsterelPlanEdge>::iterator eit = current_plan.edges.begin(); eit!=current_plan.edges.end(); eit++) {
-			for(int j=0; j<eit->sink_ids.size(); j++) {
-			for(int i=0; i<eit->source_ids.size(); i++) {
-
-				dest << "\"" << eit->source_ids[i] << "\"" << " -> \"" << eit->sink_ids[j] << "\"";
-				if(eit->duration_upper_bound == std::numeric_limits<double>::max()) {
-					dest << " [ label=\"[" << eit->duration_lower_bound << ", " << "inf]\"";
-				} else {
-					dest << " [ label=\"[" << eit->duration_lower_bound << ", " << eit->duration_upper_bound << "]\"";
-				}
-
-				// decide edge color
-				std::string edge_color = "black";
-
-				if(display_edge_type_) {
-
-					// green if conditional edge, red if start to end, blue if interference edge
-					if(eit->edge_type == rosplan_dispatch_msgs::EsterelPlanEdge::CONDITION_EDGE){
-					edge_color = "green";
-					}
-					else if(eit->edge_type == rosplan_dispatch_msgs::EsterelPlanEdge::INTERFERENCE_EDGE){
-							edge_color = "blue";
-					}
-					else if(eit->edge_type == rosplan_dispatch_msgs::EsterelPlanEdge::START_END_ACTION_EDGE){
-							edge_color = "red";
-					}
-				}
-				else {
-
-					if(edge_active[eit->edge_id]) {
-							edge_color = "red";
-					}
-					else {
-							edge_color = "black";
-					}
-				}
-
-				dest << " , penwidth=2, color=\"" << edge_color << "\"]" << std::endl;
-
-			}};
-		}
-
-		dest << "}" << std::endl;
-
-		// publish on topic
-		std_msgs::String msg;
-		msg.data = dest.str();
-		plan_graph_publisher.publish(msg);
-	}
+        node_handle = &nh;
+
+        // robust experiment parameters
+        timeout_actions = true;
+        action_timeout_fraction = 0;
+        nh.getParam("timeout_actions", timeout_actions);
+        nh.getParam("action_timeout_fraction", action_timeout_fraction);
+
+        std::string planTopic = "complete_plan";
+        nh.getParam("plan_topic", planTopic);
+        bl_end = (planTopic != "/robust_plan");
+
+        std::string plan_graph_topic = "plan_graph";
+        nh.getParam("plan_graph_topic", plan_graph_topic);
+        plan_graph_publisher = node_handle->advertise<std_msgs::String>(plan_graph_topic, 1000, true);
+
+        // display edge type with colors (conditional edge, interference edge, etc)
+        nh.param("display_edge_type", display_edge_type_, false);
+
+        reset();
+    }
+
+    EsterelPlanDispatcher::~EsterelPlanDispatcher()
+    {
+
+    }
+
+    void EsterelPlanDispatcher::reset() {
+
+        // for each node check completion, conditions, and dispatch
```

Comment:
Remove incorrect comment - better to replace it with "preempt currently executing nodes"